arduino/KnexElevator/ITask.h

```diff
@@ -0,0 +1,155 @@
+        ////////////////////////////////////////////////////////////////////////
+        struct Defaults
+        {
+            static const int Priority = tskIDLE_PRIORITY + 1;
+            static const int StackDepth = configMINIMAL_STACK_SIZE;
+        };
+
+    public:
+        ////////////////////////////////////////////////////////////////////////
+        /// @brief
+        /// Basic destructor
+        ////////////////////////////////////////////////////////////////////////
+        virtual ~ITask(void);
+
+        ////////////////////////////////////////////////////////////////////////
+        /// @brief
+        /// The name of the task.
+        ////////////////////////////////////////////////////////////////////////
+        char const* Name(void);
+
+        ////////////////////////////////////////////////////////////////////////
+        /// @brief
+        /// Initializes resources required by this task.
+        ///
+        /// @remarks
+        /// To be called from the Arduino setup() method.
+        ////////////////////////////////////////////////////////////////////////
+        virtual void Initialize(void) = 0;
+
+        ////////////////////////////////////////////////////////////////////////
+        /// @brief
+        /// Schedules the thread for execution.
+        ///
+        /// @remarks
+        /// To be called from the Arduino setup() method.
+        ////////////////////////////////////////////////////////////////////////
+        void Start(void);
+
+        ////////////////////////////////////////////////////////////////////////
+        /// @brief
+        /// Removes the thread from scheduler execution.
+        ///
+        /// @remarks
+        /// To be called from the Arduino setup() method.
+        ////////////////////////////////////////////////////////////////////////
+        void Stop(void);
+
+    protected:
+        ////////////////////////////////////////////////////////////////////////
+        /// @brief
+        /// Basic constructor
+        ////////////////////////////////////////////////////////////////////////
+        ITask(char const * taskName,
+              int taskPriority = ITask::Defaults::Priority,
+              int stackDepth = ITask::Defaults::StackDepth);
+
+        ////////////////////////////////////////////////////////////////////////
+        /// @brief
+        /// The body of the task - typically an infinite loop.
+        ///
+        /// @see RunWrapper()
+        ////////////////////////////////////////////////////////////////////////
+        virtual void Run(void) = 0;
+
+    private:
+        ////////////////////////////////////////////////////////////////////////
+        /// @brief
+        /// Thread entry point.
+        ///
+        /// @param[in] taskParameters  Task parameters set during task creation.
+        ////////////////////////////////////////////////////////////////////////
+        static void RunWrapper(void * taskParameters);
```

Comment:
Not trying to match the FreeRTOS coding standard, just camelCase for parameters.  I should really find a solid coding standard and document it.